src/Analyzer/FileVisitor.php

```diff
@@ -245,6 +245,6 @@ private function addParamDependency(Node\Param $node): void
 
     private function isScalarType(string $typeName): bool
     {
-        return \in_array($typeName, ['bool', 'int', 'float', 'string']);
+        return \in_array($typeName, ['bool', 'int', 'float', 'string', 'array']);
```

Comment:
I think we should expand this check to take into account all php's internal types (`resource`, `callable`, but also `iterable` or `Generator` and stuff like that) otherwise we are gonna face this problem again (but we should try to write a test to prove that). 

We could use [gettype](https://www.php.net/manual/en/function.gettype.php), minding that `object` is a type that can carry dependencies. Also, the method name `isScalarType` is misleading, I would rename it to `isInternalType`

Reply:
I love the idea of renaming the function to `isInternalType `
However, I tried to add all the internal types manually.
Using `gettype` can be another good solution with some differences, for example:
`gettype` returns `integer ` instead of `int` and we need to match these types of values also.

Reply:
In my original implementation I focused only on scalar types since arrays requires a special treatment: your should distinguish between typed arrays (that may have dependencies) and untyped arrays (that never have dependecies).
Maybe it can be useful to add a test with both kind of arrays.

P.S.
Another name for `isInternalType` can be `isBuiltInType` since the term "built-in" is used in the PHP documentation https://www.php.net/manual/en/language.types.intro.php

Reply:
Yeah `isBuiltInType` what do you think @micheleorselli, is it good enough at the moment?